src/main/kotlin/calculator/calculator/CalculatorImpl.kt

```diff
@@ -0,0 +1,52 @@
+package calculator.calculator
+
+import calculator.util.isOdd
+
+class CalculatorImpl : Calculator {
+    /**
+     * Node List를 순차적으로 계산한다.
+     */
+    override fun calculate(input: List<Node>): Int {
+        var indexOperator = 1
+        var indexRight = 2
+        var sum = input.getNumber(0)
+
+        while (indexRight != input.size + 1) {
+            val left = sum
+            val operator = input.getOperator(indexOperator)
+            val right = input.getNumber(indexRight)
+            sum = operator.operate(left, right)
+
+            indexRight += 2
+            indexOperator += 2
+        }
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
        var sum = input.getNumber(0)

        for (index in 1 until input.size step 2) {
            val operator = input.getOperator(index)
            val number = input.getNumber(index + 1)
            sum = operator.operate(sum, number)
        }
```

Reply:
아! 맞다 step! 😨  이제 기억났네요